File: 137-SingleNumberIi/137-SingleNumberIi.js
// Last updated: 10/03/2026, 10:03:39
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    nums.sort((a, b) => a - b);

    for (let i = 0; i < nums.length; i += 3) {
        if (nums[i] !== nums[i + 1]) {
            return nums[i];
        }
    }

    return nums[nums.length - 1];
};

File: 137-SingleNumberIi/137-SingleNumberIi.cpp
// Last updated: 03/08/2026, 10:13:15
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(int num:nums){
            mp[num]++;
        }
        for(auto i:mp){
            if(i.second == 1){
                return i.first;
            }
        }
        return -1;
    }
};